dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/LambdaEventDAO.java

```diff
@@ -39,6 +50,53 @@ public void delete(LambdaEvent lambdaEvent) {
         session.flush();
     }
 
+    private Predicate createNotNullLikeCriteria(String column, String filter, CriteriaBuilder cb, Root<LambdaEvent> event) {
+        return cb.and(cb.isNotNull(event.get(column)), cb.like(cb.upper(event.get(column)), "%" + filter.toUpperCase() + "%"));
+    }
+
+    private List<Predicate> processQuery(String filter, String sortCol, String sortOrder, CriteriaBuilder cb, CriteriaQuery query, Root<LambdaEvent> event) {
+        List<Predicate> predicates = new ArrayList<>();
+        if (!Strings.isNullOrEmpty(filter)) {
+            predicates.add(
+                // ensure we deal with null values and then do like queries on those non-null values
+                cb.or(
+                    createNotNullLikeCriteria("message", filter, cb, event), //
+                    createNotNullLikeCriteria("githubUsername", filter, cb, event), //
+                    createNotNullLikeCriteria("repository", filter, cb, event), //
+                    createNotNullLikeCriteria("type", filter, cb, event), //
+                    createNotNullLikeCriteria("reference", filter, cb, event), //
+                    createNotNullLikeCriteria("deliveryId", filter, cb, event)
+                )
+            );
+        }
+
+        if (!Strings.isNullOrEmpty(sortCol)) {
+            boolean hasSortCol = event.getModel()
+                    .getAttributes()
+                    .stream()
+                    .map(Attribute::getName)
+                    .anyMatch(sortCol::equals);
+
+            if (!hasSortCol) {
+                LOG.error(INVALID_SORTCOL_MESSAGE);
+                throw new CustomWebApplicationException(INVALID_SORTCOL_MESSAGE,
+                        HttpStatus.SC_BAD_REQUEST);
+
+            } else {
+                Path<Object> sortPath = event.get(sortCol);
+                if (!Strings.isNullOrEmpty(sortOrder) && "desc".equalsIgnoreCase(sortOrder)) {
```

Comment:
You don't need the `!Strings.isNullOrEmpty`; the `"desc".equalsIgnoreCase(sortOrder)` will work correctly if `sortOrder` is null or blank.